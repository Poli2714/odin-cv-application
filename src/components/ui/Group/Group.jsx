import PropTypes from 'prop-types';

import styles from './Group.module.css';
import ArrowDownIcon from 'src/assets/arrowDown.svg';

const size = 20;

function Group({ isActive, label, onClick }) {
  return (
    <div data-testid="group" className={styles.group} onClick={onClick}>
      <span>{label}</span>
      <img
        alt={isActive ? 'Arrow up icon' : 'Arrow down icon'}
        className={isActive ? styles.expanded : styles.collapsed}
        height={size}
        src={ArrowDownIcon}
        width={size}
      />
    </div>
  );
}

Group.propTypes = {
  isActive: PropTypes.bool,
  label: PropTypes.string,
  onClick: PropTypes.func,
};

export default Group;
