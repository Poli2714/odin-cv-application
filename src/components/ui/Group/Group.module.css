.group {
  border-bottom: 1px solid var(--clr-text-opaque);

  display: flex;
  align-items: center;

  padding: 0.5rem 0;
  width: 100%;

  & > img {
    margin-left: auto;
  }

  &:hover {
    cursor: pointer;
  }
}

.collapsed {
  transition: transform 150ms linear;
}

.expanded {
  transform: rotate(180deg);
  transition: transform 300ms linear;
}
