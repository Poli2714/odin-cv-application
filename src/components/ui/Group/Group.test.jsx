import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Group from './Group';

test('renders Group', () => {
  render(<Group isActive={false} label="Test label" onClick={() => {}} />);

  expect(screen.getByTestId('group')).toMatchSnapshot();
});

test('renders img with correct attributes when Group is active', () => {
  render(<Group isActive={true} label="Test label" onClick={() => {}} />);

  const image = screen.getByRole('img');
  expect(image.alt).toMatch(/^arrow up icon$/i);
  expect(image).toHaveClass(/^_expanded_[a-z0-9]+$/);
});

test('renders img with correct attributes when Group is inactive', () => {
  render(<Group isActive={false} label="Test label" onClick={() => {}} />);

  const image = screen.getByRole('img');
  expect(image.alt).toMatch(/^arrow down icon$/i);
  expect(image).toHaveClass(/^_collapsed_[a-z0-9]+$/);
});

test('calls event handler once when clicked on Group', async () => {
  const onClick = vi.fn();
  const user = userEvent.setup();

  render(<Group isActive={true} label="Test label" onClick={onClick} />);

  await user.click(screen.getByTestId('group'));
  expect(onClick).toHaveBeenCalledOnce();
});

test("doesn't call event handler when Group is not clicked", () => {
  const onClick = vi.fn();

  render(<Group isActive={false} label="Test label" onClick={onClick} />);

  expect(onClick).not.toHaveBeenCalled();
});
