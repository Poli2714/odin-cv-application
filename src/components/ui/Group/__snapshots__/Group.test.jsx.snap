// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Group 1`] = `
<div
  class="_group_15dcab"
  data-testid="group"
>
  <span>
    Test label
  </span>
  <img
    alt="Arrow down icon"
    class="_collapsed_15dcab"
    height="20"
    src="/src/assets/arrowDown.svg"
    width="20"
  />
</div>
`;
